### ISSUE TITLE
Finite Difference Sensitivity

### ISSUE THREAD

**Issue by lheagy** [OpenGeophysics/matrixutils#3]
Over in [opengeophysics/scratchpad](https://github.com/opengeophysics/scratchpad/), @prisae and I have been working on using SimPEG to invert empymod simulations. One piece we wrote to facilitate this is a [util to approximate the sensitivity of a function using a finite difference approximation](https://github.com/opengeophysics/scratchpad/blob/feat/plugin/plugin/utils.py#L6). This might fit in `matrixutils`? I would appreciate thoughts on scope here. cc @leouieda, @prisae, @rowanc1

**Comment by prisae** [comment]
Not sure. It wouldn't be my first thought to search a gradient function in a package called `matrixutils`. But then, the Jacobian usually is a matrix. I guess it would fit.

**Comment by prisae** [comment]
Or would it fit better into https://github.com/opengeophysics/deeplook? @leouieda

**Comment by rowanc1** [comment]
Any chance we can just rely on this:
https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.optimize.approx_fprime.html

I sort of agree with @prisae that this is weird to be in the `matrixutils` repo, seems like it fits with inversion or optimization much more.

**Comment by leouieda** [comment]
I agree that this could be probably fit in better with the inversion packages. I think PyGIMLi has something like this as well.
